### PR TITLE
Intel GPU: mode: add XPU to supported devices list

### PR DESCRIPTION
**Pull request by hjhee**
Kernel for `mode` Op is being ported to https://github.com/intel/torch-xpu-ops/pull/770, this requires adding XPU to supported device type.

Additional context: https://github.com/intel/torch-xpu-ops/issues/327

@fengyuan14 @EikanWang 

cc @frank-wei @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @gujinghui @EikanWang @fengyuan14 @guangyey

### CODE REVIEW

File: aten/src/ATen/native/TensorCompare.cpp [hunk 1]
```diff
@@ -584,8 +584,8 @@ std::tuple<Tensor, Tensor> mode(const Tensor& self, int64_t dim, bool keepdim) {
 
 std::tuple<Tensor &,Tensor &> mode_out(const Tensor& self, int64_t dim, bool keepdim,
                                        Tensor& values, Tensor& indices) {
-  TORCH_CHECK(self.device().is_cpu() || self.is_cuda(),
-              "mode only supports CPU AND CUDA device type, got: ", self.device().type());
+  TORCH_CHECK(self.device().is_cpu() || self.is_cuda() || self.is_xpu(),
+              "mode only supports CPU, CUDA, XPU device type, got: ", self.device().type());
```
Comment (fengyuan14) [inline aten/src/ATen/native/TensorCompare.cpp:588]:
```suggestion
              "mode only supports CPU, CUDA AND XPU device type, got: ", self.device().type());
```

**Comment by EikanWang** [comment]
Will the test regard the changes be covered by torch-xpu-ops?

**Review by EikanWang — DISMISSED** [review]

File: aten/src/ATen/native/TensorCompare.cpp [hunk 2]
```diff
@@ -584,8 +584,8 @@ std::tuple<Tensor, Tensor> mode(const Tensor& self, int64_t dim, bool keepdim) {
 
 std::tuple<Tensor &,Tensor &> mode_out(const Tensor& self, int64_t dim, bool keepdim,
                                        Tensor& values, Tensor& indices) {
-  TORCH_CHECK(self.device().is_cpu() || self.is_cuda(),
-              "mode only supports CPU AND CUDA device type, got: ", self.device().type());
+  TORCH_CHECK(self.device().is_cpu() || self.is_cuda() || self.is_xpu(),
```
Comment (EikanWang) [inline aten/src/ATen/native/TensorCompare.cpp:587]: Regarding the device check, I'm wondering if we should add a tensor method (`is_accelerator`) to the tensor class. @albanD , thoughts?

**Comment by fengyuan14** [comment]
> Will the test regard the changes be covered by torch-xpu-ops?

Yes, we will have ported PyTorch cases in torch-xpu-ops. The PR is a dependency for us to enable the operator on XPU. The cases in torch-xpu-ops will cover the operator for XPU, once both the PR and the PR of mode implementation in torch-xpu-ops are done.

File: aten/src/ATen/native/TensorCompare.cpp (hunk 2 above)
Comment (malfet) [reply aten/src/ATen/native/TensorCompare.cpp:587]: I think it's a good idea to generalize it

**Review by malfet — DISMISSED** [review]

**Comment by fengyuan14** [comment]
The PR was merged.

**Pull request closed without merging**